### ISSUE TITLE
[FR]: merge tar files with "deps"

### ISSUE THREAD

**Issue by gfrankliu** [aspect-build/bazel-lib#870]
### Describe the feature

In `pkg_tar`, we can use "deps" to merge list of tar files into one. 
How can we do it in `tar`?

**Comment by gfrankliu** [comment]
`pkg_tar` has a [bug](https://github.com/bazelbuild/rules_pkg/issues/873) with regard to the directory symlinks when merging/flattening multiple tar files. I am hoping aspect `tar` can handle this use case properly.

**Comment by jjmaestro** [comment]
@gfrankliu not sure if you've seen #833 and/or [`flatten` in `rules_distroless`](https://github.com/GoogleContainerTools/rules_distroless/blob/5f855ed94ac2eacc9c4212e7081eaec10214af93/distroless/private/flatten.bzl), you could probably use that.

**Comment by gfrankliu** [comment]
Yes, I tried that, but the "bug" is also there. The main [second pat of the issue](https://github.com/bazelbuild/rules_pkg/issues/873#issuecomment-2185331749) is still there.

Further discussions are [here](https://github.com/GoogleContainerTools/rules_distroless/issues/68) but unfortunately it was closed since it is too hard to fix.
